06.bowling_object/game.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require './frame'
+require './shot'
+
+class Game
+  def initialize(pinfall_text)
+    @frames = parse_pinfall_text(pinfall_text).map.with_index { |shots, index| Frame.new(index, shots) }
+  end
+
+  def main
+    total_score = @frames.sum { |frame| frame.calculate_score(@frames) }
+    puts total_score
+  end
+
+  private
+
+  def parse_pinfall_text(pinfall_text)
+    all_shots = pinfall_text.split(',').map { |pin| Shot.new(pin) }
```

Comment:
```suggestion
    all_shots = pinfall_text.split(',').map { |pinfall| Shot.new(pinfall) }
```

pinfall_textをパースしてできあがるのはpinfallかな。
また、pinはピンそのものを指しますが、pinfallには「ピンを倒した本数」という意味があるようなので、こっちの方がいいかもです。

> (bowling) The number of pins toppled by a bowler 
https://ejje.weblio.jp/content/pinfall